client/src/search/Typeahead/Typeahead.js

```diff
@@ -332,74 +322,26 @@ export class Typeahead extends React.Component {
   }
 
   get derived_menu_state() {
```

Comment:
`derived_menu_state` doesn't really feel necessary anymore (yay! things are much cleaner now), maybe we don't need it?

Reply:
Especially `total_matching_results === total_menu_items`, anywhere using either of these can just use `matching_results.length` now 🎉